app/models/ability.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
```

Comment:
Right, the main bulk of the permissions.
Essentially cancancan automatically calls Ability.new(current_user) for each request, and uses that to resolve its permissions. This basic class grants no permissions, instead other abilities get merged in.
All users get granted Ability::BaseUsers, and then subsequent privileges get ranted for each role the user has